### ISSUE TITLE
Ensure positive index is not chosen in negatives 

### ISSUE THREAD

**Issue by senthilps8** [zhirongw/lemniscate.pytorch#13]
Shouldn't you ensure  `idx != y` here? 
This probably is not a big deal for ImageNet-scale datasets, but might cause issues in smaller datasets. 
https://github.com/zhirongw/lemniscate.pytorch/blob/34c7ce367c41ea2bcc38d51087dff4fcc54f7443/lib/NCEAverage.py#L88

**Comment by zhirongw** [comment]
@senthilps8 , I guess you are absolutely correct. The code here is not exactly right.
In smaller dataset, sampling is actually not needed. You can switch to the full softmax loss as in Cifar10.

**Comment by HobbitLong** [comment]
@senthilps8 , according to my observation, this does not affect the final performance.

**Comment by senthilps8** [comment]
@HobbitLong Thanks for confirming.
